pkg/skaffold/debug/cnb.go

```diff
@@ -0,0 +1,144 @@
+	metadataJSON, found := ic.labels["io.buildpacks.build.metadata"]
+	if !found {
+		return ContainerDebugConfiguration{}, "", fmt.Errorf("image is missing buildpacks metadata; perhaps built with older lifecycle?")
+	}
+	json.Unmarshal([]byte(metadataJSON), &m)
+	if len(m.Processes) == 0 {
+		return ContainerDebugConfiguration{}, "", fmt.Errorf("buildpacks metadata has no processes")
+	}
+
+	// the buildpacks launcher is retained as the entrypoint
+	ic, rewriter := adjustCommandLine(m, ic)
+	c, img, err := transformer(container, ic)
+	if err == nil && rewriter != nil {
+		container.Args = rewriter(container.Args)
+	}
+	return c, img, err
+}
+
+// adjustCommandLine resolves the launch process and then rewrites the command-line to be
+// in a form suitable for the normal `skaffold debug` transformations.  It returns an
+// amended configuration with a function to re-transform the command-line to the form
+// expected by the launcher.
+func adjustCommandLine(m cnb.BuildMetadata, ic imageConfiguration) (imageConfiguration, func([]string) []string) {
+	// direct exec
+	if len(ic.arguments) > 0 && ic.arguments[0] == "--" {
+		// strip and restore the "--"
+		ic.arguments = ic.arguments[1:]
+		return ic, func(transformed []string) []string {
+			return append([]string{"--"}, transformed...)
+		}
+	}
+
+	processType := "web" // default buildpacks process type
+	// the launcher accepts the first argument as a process type
+	if len(ic.arguments) == 1 {
+		processType = ic.arguments[0]
+	} else if value := ic.env["CNB_PROCESS_TYPE"]; len(value) > 0 {
+		processType = value
+	}
+
+	for _, p := range m.Processes {
+		if p.Type == processType {
+			var rewriter func(transformed []string) []string
+			if p.Direct {
+				// p.Command is the command and p.Args are the arguments
+				ic.arguments = append([]string{p.Command}, p.Args...)
+				rewriter = func(transformed []string) []string {
+					return append([]string{"--"}, transformed...)
+				}
+			} else {
+				// Script type: split p.Command, pass it through the transformer, and then reassemble in the rewriter.
+				if args, err := shell.Split(p.Command); err == nil {
+					ic.arguments = args
+				} else {
+					ic.arguments = []string{p.Command}
+				}
+				rewriter = func(transformed []string) []string {
+					// reassemble back into a script with arguments
+					return append([]string{shJoin(transformed)}, p.Args...)
+				}
+			}
+			return ic, rewriter
+		}
+	}
+
+	if len(ic.arguments) == 0 {
+		// indicates an image mis-configuration as we should have resolved the the
+		// CNB_PROCESS_TYPE (if specified) or `web`.
+		logrus.Warnf("no CNB launch found for %s/%s", ic.artifact, processType)
+		return ic, nil
+	}
+
+	// ic.arguments[0] is a shell script:  split it, pass it through the transformer, and then reassemble in the rewriter.
+	var rewriter func(transformed []string) []string
+	if args, err := shell.Split(ic.arguments[0]); err == nil {
```

Comment:
What about the error case?

Reply:
There's not much we can do, but I'll add a comment.